src/aiperf/timing/fixed_schedule_strategy.py

```
@@ -26,29 +35,60 @@ def __init__(
+        # Reconstruct the full schedule from the first turn timestamps
+        self._schedule: list[tuple[int | float, str]] = []
+        for conversation in dataset_metadata.conversations:
+            if conversation.turns[0].timestamp_ms is not None:
+                # Add first turn only, as the credit is for the whole conversation
+                self._schedule.append(
+                    (conversation.turns[0].timestamp_ms, conversation.conversation_id)
+                )
+            else:
+                raise ValueError(
+                    f"Conversation {conversation.conversation_id} has no timing data"
+                )
```

Ruff (0.14.5)

Avoid specifying long messages outside the exception class

(TRY003)